src/helpers.js

```diff
@@ -195,6 +204,230 @@ module.exports = ({ types: t }) => {
     );
   };
 
+  const getAmdFactoryArgsMapper = () => {
+    // Returns the factory args mapper function.
+    // Generated code:
+    // function(dep) {
+    //   return {
+    //     require: require,
+    //     module: module,
+    //     exports: module.exports
+    //   }[dep] || require(dep);
+    // }
+    const moduleIdent = t.identifier(MODULE);
+    const requireIdent = t.identifier(REQUIRE);
+    const exportsIdent = t.identifier(EXPORTS);
+    const DEP = 'dep';
+    return t.functionExpression(
+      null,
+      [t.identifier(DEP)],
+      t.blockStatement([
+        t.returnStatement(
+          t.logicalExpression(
+            '||',
+            t.memberExpression(
+              t.objectExpression([
+                t.objectProperty(requireIdent, requireIdent),
+                t.objectProperty(moduleIdent, moduleIdent),
+                t.objectProperty(exportsIdent, t.memberExpression(moduleIdent, exportsIdent)),
+              ]),
+              t.identifier(DEP),
+              true
+            ),
+            t.callExpression(t.identifier(REQUIRE), [t.identifier(DEP)])
+          )
+        ),
+      ])
+    );
+  };
+
+  const createFactoryInvocationWithUnknownArgTypes = (
+    path,
+    opts,
+    dependencyList,
+    factory,
+    isDefineCall,
+    arity
+  ) => {
+    const factoryIdentifier = getUniqueIdentifier(path.scope, MAYBE_FUNCTION);
+    const depsIdentifier = getUniqueIdentifier(path.scope, AMD_DEPS);
+    const blockStatements = [];
+
+    // Define block scoped variables 'maybeFunction' and 'amdDeps'.
+    blockStatements.push(
+      t.variableDeclaration('var', [t.variableDeclarator(factoryIdentifier, factory)])
+    );
+    blockStatements.push(
+      t.variableDeclaration('var', [t.variableDeclarator(depsIdentifier, dependencyList)])
+    );
+
+    // If we don't know that the dependency list is an array, then we need to check
+    // the type at runtime.
+    if (!t.isArrayExpression(dependencyList)) {
+      if (isDefineCall) {
+        // If the arity of the define call is 2, then the dependency list argument
+        // could be the module name if the type is string.  In that case, we ignore
+        // the module name and use the default dependencies per the spec:
+        // https://github.com/amdjs/amdjs-api/wiki/AMD#dependencies-
+        // Generated code:
+        // if (typeof amdDeps === 'string') {
+        //   amdDeps = ['require', 'exports', 'module'];
+        // }
+        if (arity === 2) {
```

Comment:
It needs to be a separate conditional because of the else clause on the `if (isDefineCall)`.  If the conditionals are combined, as you suggest, then it changes the logic and we'd need to change the `else` to `else if (!isDefineCall)` which I don't think is any better.